src/filter.rs

```diff
@@ -54,7 +61,7 @@ impl Filter for Simple {
         match self.keys.clone() {
             None => true,
             Some(re) => {
-                let key = unsafe{str::from_utf8_unchecked(key)};
+                let key = unsafe { str::from_utf8_unchecked(key) };
```

Comment:
I realize this is my own code, but this seems ... dangerous. I don't think we control the input and so it could very well be an invalid string.

I'd recommend just using `str::from_utf8` and unwrapping that. That's safe and will simply panic instead of causing undefined behavior.
There's only 3 instances of `unsafe` in the code and all of them can probably be converted.

Reply:
Right, I didn't touch those parts myself, this was the linter. Will see that I factor out unsafe completely!